tests/unit/plugins/inventory/test_flightctl.py

```
@@ -0,0 +1,524 @@
+        device = {
+            'metadata': {},
+            'status': {'systemInfo': {'hostname': 'hostname-device'}},
+            'hostname': 'hostname-device'
+        }
+        device_id, metadata = _validate_device(device)
+        self.assertEqual(device_id, 'hostname-device')
+        self.assertEqual(metadata, {})
+
+        # Test with invalid device (no name)
+        device = {'metadata': {}}
+        with self.assertRaises(Exception):
```

Ruff (0.8.2)

`assertRaises(Exception)` should be considered evil

(B017)